tensorflow/core/kernels/xent_op.h

```diff
@@ -84,10 +93,12 @@ struct XentEigenImpl {
 #endif
 
     // max_logits along classes.
-    scratch.reshape(batch_only).device(d) = logits.maximum(along_class);
+    scratch.reshape(batch_only).device(d) =
```

Comment:
Could you please verify that the presence of the extra broadcast expression does not affect performance in the trivial case? If it does, perhaps we should have a separate code path for that. This is a pretty performance critical op for many models.

Reply:
Thanks @rmlarsen. I added zero dimensions and single classes for benchmark. No significant performance changes so far:

Before:
```
Benchmark: zero_dimension_m_128_n_100000_k_100000_use_gpu_False 	 wall_time: 0.000133 s 	 Throughput: 96.2 GB/s

Benchmark: single_class_m_128_n_100000_k_100000_use_gpu_False 	 wall_time: 0.000131 s 	 Throughput: 97.6 GB/s

```

After:
```
Benchmark: zero_dimension_m_128_n_100000_k_100000_use_gpu_False 	 wall_time: 0.000135 s 	 Throughput: 94.7 GB/s

Benchmark: single_class_m_128_n_100000_k_100000_use_gpu_False 	 wall_time: 0.000133 s 	 Throughput: 96.4 GB/s
```